src/main/java/SoloTech/StockFlow/common/aop/CacheAspect.java

```diff
@@ -74,14 +79,34 @@ public Object handleCaching(ProceedingJoinPoint joinPoint, Cached cached) throws
 
             case WRITE -> {
                 result = joinPoint.proceed();
-                if (result != null || cached.cacheNull()) {
-                    redisTemplate.opsForValue().set(cacheKey, result, Duration.ofSeconds(ttl));
+
+                if ((result != null || cached.cacheNull())
```

Comment:
```suggestion
                if (shouldCache(result, cached) && isTxActive()) { ... }
               
private boolean shouldCache(Object result, Cached cached) {
    return result != null || cached.cacheNull();
}

private boolean isTxActive() {
    return TransactionSynchronizationManager.isSynchronizationActive();
}
```

조건문이 반복되는데요 함수로 빼면 가독성이 더 좋아질것 같아요